packages/playwright-driver/main.js

```diff
@@ -17,23 +17,61 @@
 const path = require('path');
 const fs = require('fs');
 const os = require('os');
+const crypto = require('crypto');
 const util = require('util');
 
+const readFileAsync = util.promisify(fs.readFile);
+const writeFileAsync = util.promisify(fs.writeFile)
+const mkdirAsync = util.promisify(fs.mkdir)
+
+function sha1(data) {
+  const sum = crypto.createHash('sha1');
+  sum.update(data);
+  return sum.digest('hex');
+}
+
 (async () => {
+  if (process.argv.length === 2 || process.argv.some(arg => arg.includes("--help"))) {
+    console.log(`Usage:
+      './driver --print-api' - Prints the Playwright API to the stdout
+      './driver --install'   - Downloads the Playwright browsers
+      './driver --run'       - Executes the Playwright RPC server
+    `)
+    return
+  }
+  // Playwright needs to launch the PrintDeps.exe which is embedded into the NPM package.
+  // Since it's packed with PKG, we have to move it out and set the new path as an environment
+  // variable so Playwright can use it.
   if (os.platform() === 'win32') {
-    const checkDbPath = path.join(__dirname, 'node_modules', 'playwright', 'bin', 'PrintDeps.exe')
+    const printDepsPath = path.join(__dirname, '..', '..', 'bin', 'PrintDeps.exe')
+
+    const printDepsFile = await readFileAsync(printDepsPath);
+    const pwPrintDepsPath = path.join(os.tmpdir(), 'ms-playwright-print-deps.exe')
+    await writeFileAsync(pwPrintDepsPath, printDepsFile)
 
-    const content = await util.promisify(fs.readFile)(checkDbPath);
-    const output = path.join(os.tmpdir(), 'ms-playwright-print-deps.exe')
-    await util.promisify(fs.writeFile)(output, content)
+    process.env.PW_PRINT_DEPS_WINDOWS_EXECUTABLE = pwPrintDepsPath
+  }
+
+  if (process.argv[2] === "--print-api") {
+    console.log(((await readFileAsync(path.join(__dirname, 'api.json'))).toString()))
+    return
+  }
 
-    process.env.PW_PRINT_DEPS_WINDOWS_EXECUTABLE = output
+  // Create a temporary directory named by the browsers.json hash to prevent version
+  // collisions of Playwright on the local system.
+  const browsersJSON = await readFileAsync(path.join(__dirname, '..', '..', 'browsers.json'))
+  const driverDir = path.join(os.tmpdir(), "ms-playwright-driver" + sha1(browsersJSON).slice(0, 5))
```

Comment:
I don't think we should pollute the fs and this will make the browser indefinitely permanent. I'd suggest that you dump browserJSON to the process.cwd() and run it from within the pip wheel folder. Would that work?

Reply:
Would probably work. And in languages where's no dependency folder available a tempdir could be used.